src/services/services.ts

```diff
@@ -4616,7 +4640,12 @@ namespace ts {
                 }
 
                 let signature: Signature;
-                type = isThisExpression ? typeChecker.getTypeAtLocation(location) : typeChecker.getTypeOfSymbolAtLocation(symbol, location);
+                type = isThisExpression
+                    ? typeChecker.getTypeAtLocation(location)
+                    : isConstructor
+                        // For constructor, get type of the class.
+                        ? typeChecker.getTypeOfSymbolAtLocation(symbol.parent, location)
+                        : typeChecker.getTypeOfSymbolAtLocation(symbol, location);

```

Comment:
When I hover over `constructor` in a class named `K` I get a display like "constructor K(n: number): K", but when I hover over `K`, I get "class K". Is that the old behaviour?